velox/functions/prestosql/aggregates/MinMaxAggregates.cpp

```diff
@@ -361,6 +361,39 @@ class NonNumericMinMaxAggregateBase : public exec::Aggregate {
   }
 
  protected:
+  template <typename TCompareTest>
+  void updateAccumulator(
+      SingleValueAccumulator* accumulator,
+      DecodedVector& decoded,
+      vector_size_t i,
+      TCompareTest compareTest) {
+    // ARRAY comparison not supported for arrays with null elements
+    // ROW comparison not supported for fields with null elements
+    // MAP comparison not supported for null value elements
+    auto indices = decoded.indices();
+    auto baseVector = decoded.base();
+    VELOX_CHECK_NOT_NULL(accumulator);
+    if (!accumulator->hasValue()) {
+      accumulator->write(baseVector, indices[i], allocator_);
+      return;
+    }
+
+    static const CompareFlags kCompareFlags{
+        true, // nullsFirst
+        true, // ascending
+        false, // equalsOnly
+        CompareFlags::NullHandlingMode::StopAtNull};
```

Comment:
min()/max() for Varchar and Varbinary are also implemented through NonNumericMinMaxAggregateBase. I wonder whether they are supposed to StopAtNull or not.

Reply:
@kagamiori StopAtNull affects only complex types. Doesn't affect Varchar and Varbinary .

Reply:
Ah I see. This makes sense.